### PR TITLE
Decompile various battle_anim files

### PR DESCRIPTION
**Pull request by progreon**

### CODE REVIEW

File: src/battle/anim/wisp_fire.c [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+    sprite->data[3] += 0xC0 * 2;
+    sprite->data[4] += 0xA0;
+
+    sprite->pos2.x = Sin(sprite->data[1], sprite->data[3] >> 8);
+    sprite->pos2.y = Cos(sprite->data[1], sprite->data[4] >> 8);
+
+    sprite->data[1] = (sprite->data[1] + 7) & 0xFF;
+
+    if (gMain.inBattle)
```
Comment (camthesaxman) [inline src/battle/anim/wisp_fire.c]: Please put braces here. You have a dangling else.

File: src/battle/anim/wisp_fire.c [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+        if ((u16) (sprite->data[1] - 0x40) > 0x83)
```
Comment (camthesaxman) [inline src/battle/anim/wisp_fire.c]: This is equivalent to `if (sprite->data[1] < 64 || sprite->data[1] > 195)`.

Comment (progreon) [reply src/battle/anim/wisp_fire.c]: Oh wow, totally didn't think about that, thanks!

File: src/battle/anim/bug.c [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+    sprite->data[5] = gBattleAnimArgs[3];
+    sprite->callback = sub_80DCA38;
+}
+
+static void sub_80DCA38(struct Sprite *sprite)
+{
+    if (sub_8078B5C(sprite))
+    {
+        DestroyAnimSprite(sprite);
+        return;
```
Comment (progreon) [inline src/battle/anim/bug.c]: Is this return fine, or do you rather have an if-else?
Destroy + return has to be used in another function of another file so I continued using it like that without realising.

File: asm/bug.s [hunk 4]
```diff
@@ -7,285 +7,285 @@
...
+# 	ldrh r0, [r4, 0x3A]
+# 	adds r0, 0xD
+# 	movs r1, 0xFF
+# 	ands r0, r1
+# 	strh r0, [r4, 0x3A]
+# _080DCA6A:
+# 	pop {r4}
+# 	pop {r0}
+# 	bx r0
+# 	thumb_func_end sub_80DCA38
```
Comment (ProjectRevoTPP) [inline asm/bug.s]: please remove after you decompile

**Comment by progreon** [comment]
This may be reviewed and merged for now, will continue this in a few days.

**Pull request merged**